pkg/detectors/coinbase_waas/coinbase_waas.go

```diff
@@ -0,0 +1,142 @@
+package coinbase_waas
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"github.com/coinbase/waas-client-library-go/auth"
+	"github.com/coinbase/waas-client-library-go/clients"
+	v1clients "github.com/coinbase/waas-client-library-go/clients/v1"
+	v1 "github.com/coinbase/waas-client-library-go/gen/go/coinbase/cloud/pools/v1"
+	"github.com/google/uuid"
+	"google.golang.org/api/googleapi"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	// Reference: https://docs.cloud.coinbase.com/waas/docs/auth
+	keyNamePat = regexp.MustCompile(`(organizations\\*/\w{8}-\w{4}-\w{4}-\w{4}-\w{12}\\*/apiKeys\\*/\w{8}-\w{4}-\w{4}-\w{4}-\w{12})`)
+	privKeyPat = regexp.MustCompile(`(-----BEGIN EC(?:DSA)? PRIVATE KEY-----(?:\r|\n|\\+r|\\+n)(?:[a-zA-Z0-9+/]+={0,2}(?:\r|\n|\\+r|\\+n))+-----END EC(?:DSA)? PRIVATE KEY-----(?:\r|\n|\\+r|\\+n)?)`)
+
+	nameReplacer = strings.NewReplacer("\\", "")
+	keyReplacer  = strings.NewReplacer(
+		"\r\n", "\n",
+		"\\r\\n", "\n",
+		"\\n", "\n",
+		"\\r", "\n",
+	)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"organizations", "apiKeys", "ec"}
```

Comment:
@rgmz should `ec` be `begin ec`?

Reply:
Probably. I wasn't sure how spaces or letters within other words would be handled (i.e., would `begin ec` catch both `BEGIN EC ` and `BEGIN ECDSA`)?

Though I suppose you could just add both as separate keywords.

Reply:
`begin ec` seems to work for both cases.

![image](https://github.com/trufflesecurity/trufflehog/assets/32133502/8647e21b-e01d-40ff-b5a8-d46c33cf68ae)